### ISSUE TITLE
View::__toString() must not throw an exception

### ISSUE THREAD

**Issue by mxmtsk** [aryehraber/statamic-responsive-img#15]
I'm trying to display an image from a bard field with the following code:

```{{ responsive_img:image }}```

Serve Cached Images directly **is enabled** and when opening the page it does generate the images in the right folder. I still get the following error.

> Method Illuminate\View\View::__toString() must not throw an exception, caught ErrorException: file_get_contents(/work/blog/https://myblog.test/img/containers/main/./office-view.jpg/0798d4bb7bb33faeb1795ccd290e88f4.jpg): failed to open stream: No such file or directory (View: /work/blog/site/addons/ResponsiveImg/resources/views/img.blade.php)

As highlighted, somehow the domain name `https://myblog.test` get's added in between the path which i think could be the error. I don't know why this happens. Am I missing something? I'm running the page locally on Laravel Valet

**Comment by simonhamp** [comment]
I'm seeing a similar issue when using multiple locales, which all seems to be related to adjusting the `$site_root` variable and the locale `url` route.

**Comment by simonhamp** [comment]
I've managed to patch this by adding the following to `ResponsiveImg.php getTinyImage()`:

```
$path = $this->getManipulatedImage(['w' => 32, 'blur' => 8]);

// Add this line between line 116 and 118
$path = str_replace('/' . strtolower(Config::getLocaleName()), '', $path);

$image = file_get_contents(webroot_path(urldecode($path)));
```

**Comment by aryehraber** [comment]
@simonhamp Thanks for looking into this, I'll try and give your above change a test locally to ensure there aren't any breaking changes and push a patch update if all looks good.

**Comment by simonhamp** [comment]
Slightly better version less likely to break URLs:

```php
use Stringy\StaticStringy;

class ResponsiveImg
{
    protected function getTinyImage()
    {
        ...
        $locale = '/' . snake_case(camel_case(strtolower(Config::getFullLocale())), '-');
        $path = StaticStringy::removeLeft(strtolower($path), $locale);
        ...
    }
}
```

**Comment by NA-Dev** [comment]
I still have the error with this "fix" after clearing the cache.

Method Illuminate\View\View::__toString() must not throw an exception, caught ErrorException: file_get_contents(/Users/nikki/Sites/website/https://website.test/img/containers/main/./image.png/776b963fe18cd78275a01fac2f66ffb2.jpg): failed to open stream: No such file or directory (View: /Users/nikki/Sites/website/site/addons/ResponsiveImg/resources/views/img.blade.php)

**Comment by potentweb** [comment]
On line 125
 
`return $this->image->manipulate(array_merge($default, $params));`

Returns a path with the locale url prepended. It works ok if the url is '/' but otherwise breaks line 116.

I changed the return (line 125) to strip off the domain

`return parse_url($this->image->manipulate(array_merge($default, $params)))['path'];`

This also makes all the img srcs relative again.

**Comment by websmyth** [comment]
This worked for me, too, @potentweb. ResponsiveImg was working on one local site but not another, the only difference being a locale in the system.yaml. Removing that fixed the issue.

**Comment by aryehraber** [comment]
Hi all! Firstly, apologies for the very long wait on this, I just couldn't quite crack it in my early attempts and then didn't have the time to look more closely... until now. 

Thanks for all the suggestions and workaround fixes; ended up with a slightly different solution but it all helped to get to the bottom of it.

Just released the new version to Statamic's Marketplace [download here](https://statamic.com/marketplace/addons/responsive-img), please open a new issue if you continue seeing this pop up. Will hopefully not take me another 10 months to get to it 🤦‍♂️

**Comment by simonhamp** [comment]
This is still an issue for us. We get duplicated locale parts of the URL in the image path.

Our locale config looks like this (which works fin in Statamic):

```yaml
locales:
  en:
    full: en
    name: 'Global (English)'
    url: /
  en_US:
    full: en_US
    name: 'United States'
    url: /en-us/
```
However, this config appears to be problematic with this addon.

If we make the `url` parameters absolute URLs and remove the call to `URL::makeRelative` that you've added in  `ResponsiveImg::getManipulatedImage`, it works for us.

It feels like this `makeRelative` call is redundant, as when you then call `[Str::removeLeft](https://github.com/aryehraber/statamic-responsive-img/blob/master/ResponsiveImg/ResponsiveImg.php#L117)` it can't possibly remove the root URL (which is absolute) from the now-relative image path.

**Comment by aryehraber** [comment]
Thanks for following up @simonhamp! Will try and revisit this on a multi-locale setup.

**Comment by aryehraber** [comment]
@simonhamp I was finally able to reproduce the issue using a multi-locale setup. I _think_ I have a better fix now, but would be great if you can give it a test run before I officially release this.

You can download the branch directly here:
https://github.com/aryehraber/statamic-responsive-img/archive/fix/locale-img-path.zip

Thanks!

**Comment by aryehraber** [comment]
Going ahead and merging the above fix which _seemed_ to work well in my testing. Please open a new GH Issue if there are further issues relating to this!

**Comment by simonhamp** [comment]
@fbenas looks like we need to try this out!
